homeassistant/components/devolo_home_network/coordinator.py

```diff
@@ -0,0 +1,44 @@
+"""Base coordinator."""
+
+from asyncio import Semaphore
+from collections.abc import Awaitable, Callable
+from datetime import timedelta
+from logging import Logger
+
+from typing_extensions import TypeVar
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_DataT = TypeVar("_DataT")
+
+
+class DevoloDataUpdateCoordinator(DataUpdateCoordinator[_DataT]):
+    """Class to manage fetching data from devolo Home Network devices."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        logger: Logger,
+        *,
+        name: str,
+        semaphore: Semaphore,
+        update_interval: timedelta | None = None,
+        update_method: Callable[[], Awaitable[_DataT]] | None = None,
```

Comment:
Not a full review, but I don't think you need to have defaults here.
```suggestion
        update_interval: timedelta,
        update_method: Callable[[], Awaitable[_DataT]],
```